maint.php

```diff
@@ -819,57 +845,235 @@ function applyFilter() {
+					<td>
+						<select id='site_id'>
+							<option value='<?php print HOST_FILTER_ANY ?>' <?php if (get_request_var('site_id') == HOST_FILTER_ANY) {?> selected<?php }?>><?php print __('Any');?></option>
+							<option value='<?php print HOST_FILTER_NONE ?>' <?php if (get_request_var('site_id') == HOST_FILTER_NONE) {?> selected<?php }?>><?php print __('None');?></option>
+							<?php
+							$sites = db_fetch_assoc('SELECT id, name
+								FROM sites
+								WHERE id IN (SELECT site_id FROM host)
+								ORDER BY name');
+							if (cacti_sizeof($sites)) {
+								foreach ($sites as $site) {
+									print "<option value='" . $site['id'] . "'"; if (get_request_var('site_id') == $site['id']) { print ' selected'; } print '>' . html_escape($site['name']) . "</option>";
+								}
+							}
+							?>
+						</select>
+					</td>
+					<td>
+						<?php print __('Data Collector');?>
+					</td>
+					<td>
+						<select id='poller_id'>
+							<option value='<?php print HOST_FILTER_ANY ?>' <?php if (get_request_var('poller_id') == HOST_FILTER_ANY) {?> selected<?php }?>><?php print __('Any');?></option>
+							<?php
+							$pollers = db_fetch_assoc("SELECT id, name
+								FROM poller
+								ORDER BY name");
+
+							if (cacti_sizeof($pollers)) {
+								foreach ($pollers as $poller) {
+									print "<option value='" . $poller['id'] . "'"; if (get_request_var('poller_id') == $poller['id']) { print ' selected'; } print '>' . html_escape($poller['name']) . "</option>";
+								}
+							}
+							?>
+						</select>
 					</td>
 					<td>
-						<input type='text' id='filter' size='25' value='<?php print htmlspecialchars(get_request_var('filter'));?>' onChange='applyFilter()'>
+						<?php print __('Location');?>
+					</td>
+					<td>
+						<select id='location'>
+							<option value='<?php print HOST_FILTER_LOC_ANY ?>' <?php if (get_request_var('location') == HOST_FILTER_LOC_ANY) {?> selected<?php }?>><?php print __('Any');?></option>
+							<?php
+							/* Filter location list based on other selections
+							 * List item None for Null and empty values
+							 * List item "Any" and "None" special values at top of list but can also be in data as another list entry for those; "None" could be listed twice*/
+
+							if (get_request_var('site_id') != HOST_FILTER_ANY) {
+								$sql_where .= ' AND site_id = ?';
+								$sql_where_params = array_merge($sql_where_params, array(get_request_var('site_id')));
+							}
+
+							if (get_request_var('poller_id') != HOST_FILTER_ANY) {
+								$sql_where .= ' AND poller_id = ?';
+								$sql_where_params = array_merge($sql_where_params, array(get_request_var('poller_id')));
+							}
+
+							if (get_request_var('associated') == 'false') {
+								/* Show all items */
+							} else {
+								$sql_where .= ' AND id IN (SELECT host FROM plugin_maint_hosts WHERE type = 1 AND schedule = ?)';
+								$sql_where_params = array_merge($sql_where_params, array(get_request_var('id')));
```

Comment:
Like the use of sql_where + sql_where_params 👍 

(might be biased as it looks based on my own code).

Reply:
Yes, I used the good examples :)